src/transformers/trainer.py

```diff
@@ -2869,6 +2871,17 @@ def _load_rng_state(self, checkpoint):
                         f"Didn't manage to set back the RNG states of the MLU because of the following error:\n {e}"
                         "\nThis won't yield the same results as if the training had not been interrupted."
                     )
+        if is_torch_musa_available():
```

Comment:
General note for @muellerzr - it looks like we could abstract out a lot of this common code :)